From 97439ede55a263b43a1630ac9a8f7b253ba41128 Mon Sep 17 00:00:00 2001
From: Beniamino Galvani <bgalvani@redhat.com>
Date: Thu, 11 Aug 2016 15:25:21 +0200
Subject: [PATCH] applet: fix handling of 802.1x secrets for ethernet
 connections

When @tmp_connection is destroyed it clears all its secrets: remove
the 802.1x setting before unreferencing the connection since the
setting is used by another connection.

Fixes: 3722a190468f5ce5898db8838611bc826d3153f1
---
 src/ethernet-dialog.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ethernet-dialog.c b/src/ethernet-dialog.c
index e5aaa84..c50ffb6 100644
--- a/src/ethernet-dialog.c
+++ b/src/ethernet-dialog.c
@@ -151,12 +151,16 @@ nma_ethernet_dialog_get_connection (GtkWidget *dialog)
 
 	/* Grab it and add it to our original connection */
 	s_8021x = nm_connection_get_setting (tmp_connection, NM_TYPE_SETTING_802_1X);
 	nm_connection_add_setting (connection, NM_SETTING (g_object_ref (s_8021x)));
 
 	/* Save new CA cert ignore values to GSettings */
 	eap_method_ca_cert_ignore_save (tmp_connection);
 
+	/* Remove the 8021x setting to prevent the clearing of secrets when the
+	 * simple-connection is destroyed.
+	 */
+	nm_connection_remove_setting (tmp_connection, NM_TYPE_SETTING_802_1X);
 	g_object_unref (tmp_connection);
 
 	return connection;
 }
-- 
2.5.5

